public/app/features/dashboard/components/DashboardSettings/AutoRefreshIntervals.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import { Input, Tooltip } from '@grafana/ui';
+import { defaultIntervals } from '@grafana/ui/src/components/RefreshPicker/RefreshPicker';
+
+import { getTimeSrv } from '../../services/TimeSrv';
+
+export interface Props {
+  renderCount: number; // hack to make sure Angular changes are propagated properly, please remove when DashboardSettings are migrated to React
+  refreshIntervals: string[];
+  onRefreshIntervalChange: (interval: string[]) => void;
+  getIntervalsFunc?: typeof getValidIntervals;
+  validateIntervalsFunc?: typeof validateIntervals;
+}
+
+export const AutoRefreshIntervals: FC<Props> = ({
+  renderCount,
+  refreshIntervals,
+  onRefreshIntervalChange,
+  getIntervalsFunc = getValidIntervals,
+  validateIntervalsFunc = validateIntervals,
+}) => {
+  const [intervals, setIntervals] = useState<string[]>(getIntervalsFunc(refreshIntervals ?? defaultIntervals));
+  const [invalidIntervalsMessage, setInvalidIntervalsMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    const intervals = getIntervalsFunc(refreshIntervals ?? defaultIntervals);
+    setIntervals(intervals);
+  }, [renderCount, refreshIntervals]);
+
+  const intervalsString = useMemo(() => {
+    if (!Array.isArray(intervals)) {
+      return '';
+    }
+
+    return intervals.join(',');
+  }, [intervals]);
+
+  const onIntervalsChange = useCallback(
+    (event: React.FormEvent<HTMLInputElement>) => {
+      const newIntervals = event.currentTarget.value ? event.currentTarget.value.split(',') : [];
+
+      setIntervals(newIntervals);
+    },
+    [setIntervals]
+  );
+
+  const onIntervalsBlur = useCallback(
+    (event: React.FormEvent<HTMLInputElement>) => {
+      const invalidMessage = validateIntervalsFunc(intervals);
+
+      if (invalidMessage === null) {
+        // only refresh dashboard JSON if intervals are valid
+        onRefreshIntervalChange(getIntervalsFunc(intervals));
+      }
+
+      setInvalidIntervalsMessage(invalidMessage);
+    },
+    [intervals, onRefreshIntervalChange, setInvalidIntervalsMessage]
+  );
+
+  return (
+    <div className="gf-form">
+      <span className="gf-form-label width-7">Auto-refresh</span>
```

Comment:
Just a thought but I wonder if we could use `label` here instead of `span`? It'd make the field more accessible (combined with added `htmlFor` and `id` to the input element) and would make it easier to query for this input in tests.

Reply:
Yepp this was on my list, but unfortunately `id` or `name` props are not honored by our Input component.